File: csharp/07 - Demo 3 - Refactoring/Refactoring.Tests/ReplaceTypeWithStrategy/EmployeePayAmountTest.cs
using Xunit;
using FluentAssertions;
using Refactoring.ReplaceTypeWithStrategy;

namespace Refactoring.Tests.ReplaceTypeWithStrategy
{
    public class EmployeePayAmountTest
    {
        private const int BONUS = 560;
        private const int COMMISSION = 200;
        private const int MONTHLY_SALARY = 3500;

        [Fact]
        public void EngineerPayAmountIsMonthlySalary()
        {
            var employee = CreateEmployee(Employee.ENGINEER);

            var result = employee.PayAmount();
            result.Should().Be(MONTHLY_SALARY);
        }

        [Fact]
        public void SalesmanPayAmountIsMonthlySalaryPlusCommision()
        {
            var employee = CreateEmployee(Employee.SALESMAN);

            var result = employee.PayAmount();
            result.Should().Be(MONTHLY_SALARY + COMMISSION);
        }

        [Fact]
        public void ManagerPayAmountIsMonthlySalaryPlusBonus()
        {
            var employee = CreateEmployee(Employee.MANAGER);

            var result = employee.PayAmount();
            result.Should().Be(MONTHLY_SALARY + BONUS);
        }

        [Fact]
        public void EngineerBusinessCardPaperQuality()
        {
            var employee = CreateEmployee(Employee.ENGINEER);

            var result = employee.BusinessCardPaperQuality();
            result.Should().Be("recycled");
        }

        [Fact]
        public void SalesmanBusinessCardPaperQuality()
        {
            var employee = CreateEmployee(Employee.SALESMAN);

            var result = employee.BusinessCardPaperQuality();
            result.Should().Be("high quality white");
        }

        [Fact]
        public void ManagerBusinessCardPaperQuality()
        {
            var employee = CreateEmployee(Employee.MANAGER);

            var result = employee.BusinessCardPaperQuality();
            result.Should().Be("superb quality with shiny finish");
        }

        private Employee CreateEmployee(int employeeType)
        {
            return new Employee(employeeType)
            {
                MonthlySalary = MONTHLY_SALARY,
                Commision = COMMISSION,
                Bonus = BONUS
            };
        }
    }
}

File: java/07 - Demo 3 - Refactoring/src/test/java/replacetypecodewithstrategy/EmployeePayAmountTest.java
package replacetypecodewithstrategy;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class EmployeePayAmountTest {

	private static final int BONUS = 560;
	private static final int COMMISION = 200;
	private static final int MONTHLY_SALARY = 3500;

	@Test
	public void engineerPayAmountIsMonthlySalary() {
		Employee employee = createEmployee(Employee.ENGINEER);
		
		int result = employee.payAmount();
		assertEquals(MONTHLY_SALARY, result);
	}
	
	@Test
	public void salesmanPayAmountIsMonthlySalaryPlusCommision() {
		Employee employee = createEmployee(Employee.SALESMAN);
		
		int result = employee.payAmount();
		assertEquals(MONTHLY_SALARY + COMMISION, result);
	}
	
	@Test
	public void managerPayAmountIsMonthlySalaryPlusBonus() {
		Employee employee = createEmployee(Employee.MANAGER);
		
		int result = employee.payAmount();
		assertEquals(MONTHLY_SALARY + BONUS, result);
	}
	
	@Test
	public void engineerBusinessCardPaperQuality() {
		Employee employee = createEmployee(Employee.ENGINEER);
		
		String result = employee.businessCardPaperQuality();
		assertEquals("recycled", result);
	}
	
	@Test
	public void salesmanBusinessCardPaperQuality() {
		Employee employee = createEmployee(Employee.SALESMAN);
		
		String result = employee.businessCardPaperQuality();
		assertEquals("high quality white", result);
	}
	
	@Test
	public void managerBusinessCardPaperQuality() {
		Employee employee = createEmployee(Employee.MANAGER);
		
		String result = employee.businessCardPaperQuality();
		assertEquals("superb quality with shiny finish", result);
	}
	
	private Employee createEmployee(int employeeType) {
		Employee employee = new Employee(employeeType);
		employee.setMonthlySalary(MONTHLY_SALARY);
		employee.setCommision(COMMISION);
		employee.setBonus(BONUS);
		return employee;
	}	
}
